auth/realm/base/src/main/java/org/wildfly/security/auth/realm/FileSystemRealmUtil.java

```diff
@@ -42,23 +44,38 @@ public class FileSystemRealmUtil {
      * @throws RealmUnavailableException if either realm is unavailable
      */
     public static void createEncryptedRealmFromUnencrypted(FileSystemSecurityRealm unencryptedRealm, FileSystemSecurityRealm encryptedRealm) throws RealmUnavailableException {
-        Assert.checkNotNullParam("unencryptedRealm", unencryptedRealm);
-        Assert.checkNotNullParam("encryptedRealm", encryptedRealm);
+        transferIdentitiesToNewRealm(unencryptedRealm, encryptedRealm);
+    }
+
+    /**
+     * Converts a pre-existing {@code FileSystemSecurityRealm} to a signed {@code FileSystemSecurityRealm} with integrity validation.
+     *
+     * @param inputFileSystemRealm the {@code FileSystemSecurityRealm} without integrity validation
+     * @param outputFileSystemRealm the {@code FileSystemSecurityRealm} configured with a key pair to validate data integrity
+     * @throws RealmUnavailableException if either realm is unavailable
+     */
+    public static void createRealmWithIntegrityValidation(FileSystemSecurityRealm inputFileSystemRealm,
+                                  FileSystemSecurityRealm outputFileSystemRealm) throws RealmUnavailableException {
+        transferIdentitiesToNewRealm(inputFileSystemRealm, outputFileSystemRealm);
```

Comment:
What is this method intended to do? It looks like both this method and `createEncryptedRealmFromUnencrypted` currently do the same thing which is just call `transferIdentitiesToNewRealm`. Perhaps the name / description of these methods just needs to be updated perhaps? The current description here says the outputFileSystemRealm is also configured with a key pair to verify the integrity but it doesn't look like there's anything specific to integrity validation here.

Reply:
Missed this, it was a leftover from when I wasn't sure if they would need separate functionality